gateway/src/http_entrypoints.rs

```diff
@@ -471,13 +505,34 @@ async fn sp_component_serial_console_detach(
 // TODO: how can we make this generic enough to support any update mechanism?
 #[derive(Deserialize, JsonSchema)]
 pub struct UpdateBody {
+    /// An identifier for this update.
+    ///
+    /// This ID applies to this single instance of the API call; it is not an
+    /// ID of `image` itself. Multiple API calls with the same `image` should
+    /// use different IDs.
+    pub id: Uuid,
```

Comment:
MGS generating the UUID was my gut feeling also, but RM and Andrew talked me out of it :joy:. I believe the reasoning was that nexus would generate the ID as part of its intent to start the update, record it to cockroach, then send that down to MGS. @andrewjstone am I remembering that correctly?

Reply:
Yes, exactly. I saw this comment come in yesterday and was waiting to pleasantly disagree with Adam today 🤣

Reply:
To add more to this, I think it's very likely that we'll want to track updates in CockroachDb and be able to query them from there. Otherwise you have to end up hitting all SPs to get a global status. If you create the SP in MGS, you could always write that UUID along with when the update finishes, but then it makes any coordination of work a bit harder. In short, I don't think it's strictly necessary to shape the API this way, but my gut tells me we'll end up doing it this way eventually.

Reply:
I see: we want to put the UUID into CRDB **before** making the initial call to MGS from Nexus, is that right?